kong.conf.default

```diff
@@ -904,6 +904,10 @@
 #nginx_admin_client_body_buffer_size = 10m  # Defines the buffer size for reading
                                             # the request body on Admin API.
 
+#nginx_http_lua_regex_match_limit = 100000  # Global MATCH_LIMIT for PCRE
+                                            # bounds most degenerate cases
+                                            # to less than 2 seconds.
```

Comment:
```suggestion
#nginx_http_lua_regex_match_limit = 100000  # Global `MATCH_LIMIT` for PCRE
                                            # regex matching. The default of `100000` should ensure
                                            # at worst any regex Kong executes could finish within
                                            # roughly 2 seconds.
```

Reply:
"any regex Kong executes could finish within roughly 2 seconds"

I think this phrasing might be a bit misleading. This ensures an upper bound in the execution time, but some regexes will finish with an error

Reply:
either an error on a false negative.   but i think it's ok; i think any expression that takes so long is wrong already.  also, in the vast majority of cases, the degenerate behaviour happens only when no match is possible, so failing to match is the right answer, even if the engine couldn't test every possible permutation.